CONTRIBUTING.md

```diff
@@ -1,35 +1,211 @@
+
+
+## Before submitting an enhancement/feature request
+
+*   Check the wiki and forum for tips — you might discover that the enhancement is already available.
+*   Most importantly, check if you're using the latest version of qBittorrent and if you can get the desired behavior by changing qBittorrent's settings.
+*   Check if there's already a beta version that enhancement.
+*   Perform a cursory search to see if the enhancement has already been suggested. If it has, add a comment to the existing issue instead of opening a new one.
+
+
+## Steps to ensure a good enhancement/feature suggestion
+
+
+-   Specify which version of qBittorrent you're using.
+-   Specify the name and version of the OS you're using.
+-   Provide a step-by-step description of the suggested enhancement in as many details as possible.
+-   Describe the current behavior and explain which behavior you expected to see instead and why.
+-   Include screenshots and animated GIFs which help you demonstrate the steps or point out the part of qBittorrent which the suggestion is related to.
+-   If this enhancement exists in other bittorrent clients, list list those clients.
 
 
 # Opening a pull request
 
 ### Must read
-* Read our [**coding guidelines**](https://github.com/qbittorrent/qBittorrent/blob/master/CODING_GUIDELINES.md). There are some scripts to help you: [uncrustify script](https://raw.githubusercontent.com/qbittorrent/qBittorrent/master/uncrustify.cfg), [astyle script](https://gist.github.com/Chocobo1/539cee860d1eef0acfa6), [(related thread)](https://github.com/qbittorrent/qBittorrent/issues/2192).
-* Keep the title **short** and provide a **clear** description about what your pull request does.
-* Provide **screenshots** for UI related changes.
-* Keep your git commit history **clean** and **precise**. Commits like `xxx fixup` should not appear.
-* If your commit fix a reported issue (for example #4134), add the following message to the commit `Closes #4134.`. Example [here](https://github.com/qbittorrent/qBittorrent/commit/a74bac20c4e8de9776bf9bb77fdc7526135d1988).
+*   Read our [**coding guidelines**](https://github.com/qbittorrent/qBittorrent/blob/master/CODING_GUIDELINES.md). There are some scripts to help you: [uncrustify script](https://raw.githubusercontent.com/qbittorrent/qBittorrent/master/uncrustify.cfg), [astyle script](https://gist.github.com/Chocobo1/539cee860d1eef0acfa6), [(related thread)](https://github.com/qbittorrent/qBittorrent/issues/2192).
+*   Keep the title **short** and provide a **clear** description about what your pull request does.
+*   Provide **screenshots** for UI related changes.
+*   Keep your git commit history **clean** and **precise**. Commits like `xxx fixup` should not appear.
+*   If your commit fix a reported issue (for example #4134), add the following message to the commit `Closes #4134.`. Example [here](https://github.com/qbittorrent/qBittorrent/commit/a74bac20c4e8de9776bf9bb77fdc7526135d1988).
```

Comment:
fix -> fixes

Reply:
An example of a bad commit message, IMO. However, we still don't have rules about it.

Reply:
Ok

Reply:
> An example of a bad commit message, IMO. However, we still don't have rules about it.

@Chocobo1 Do you have a suggestion of a better message? Your original comment is neither helpful nor constructive.

Reply:
>An example of a bad commit message, IMO. However, we still don't have rules about it.
>
>@Chocobo1 Do you have a suggestion of a better message? Your original comment is neither helpful nor constructive.

Geez, you guys need to learn respect the history, bashing it repetitively won't change anything, that was an example for **"Close #XXXX"**, NOT an example for perfect commit log.
And no, please find/write your perfect commit msg yourself.

Reply:
I'm sorry if I offended someone, my intention was not that. 
What matters is not only that one message is better than another. But the fact that we do not adhere to one specific style in this, and our history looks like a dump. I have several times hinted that it would be nice to have some "commit messaging" rules (as part of our Coding Guidelines).
[This](https://chris.beams.io/posts/git-commit/) topic contains some useful reflections on this subject, and we could take it as a basis.

Reply:
@Chocobo1 My apologies, I included you by mistake. My comment was in response to @glassez.

Reply:
@glassez 
> I have several times hinted that it would be nice to have some "commit messaging" rules (as part of our Coding Guidelines).

I see, but in that case, isn't it outside the scope of this PR?

Reply:
I just don't want to see link to commit with incorrect (IMO) message as an example of "how it should be done". Until we have "commit messaging" rules.

Reply:
I would have removed the Example, and the rest left as is.
Wait for other opinions.